### PR TITLE
Add a Custom 404 page 

### PR DESCRIPTION
**Pull request by Khizar-sheikh**
**- Adds a custom 404 error page for improved user experience on invalid routes.**

### CODE REVIEW

File: app/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import Image from "next/image";
+import SVGIMG from "./404.svg"
```
Comment (napstar-420) [inline app/not-found.tsx]: import using `@` alias

File: app/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { BackgroundGrid } from "@/components/background-grid";
+import Link from "next/link";
+import { CONFIG } from "./config";
```
Comment (napstar-420) [inline app/not-found.tsx]: import using `@` alias

File: app/not-found.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export default function NotFound() {
+  return (
+    <div className="flex h-screen flex-col lg:flex-row items-center justify-center">
+        <BackgroundGrid/>
+      {/* Left Half */}
+      <div className="flex flex-col items-center justify-center text-center p-8 lg:w-1/2">
+        <h1 className="text-6xl font-bold text-[#34CB85]">404</h1>
```
Comment (napstar-420) [inline app/not-found.tsx]: to use brand color use `brand-primary`, so the text class will become `text-brand-primary`

File: app/not-found.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <h2 className="text-2xl font-semibold mt-4 text-gray-800 dark:text-white">
```
Comment (napstar-420) [inline app/not-found.tsx]: don't use colors like `text-gray-800 dark:text-white` cause then we will have to give colors for both light and dark mode, use the css variables declared in global.css, e.g. `primary`, `primary-foreground`, `secondary` etc

File: app/not-found.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          Page Not Found
+        </h2>
+        <p className="mt-2 text-gray-600 dark:text-white">
```
Comment (napstar-420) [inline app/not-found.tsx]: same here

**Review by Khizar-sheikh — COMMENTED** [review]
Adds a custom 404 error page for improved user experience on invalid routes.

**Review by napstar-420 — APPROVED** [review]

**Pull request merged**